mesonbuild/msubprojects.py

```diff
@@ -435,6 +439,41 @@ def update(self) -> bool:
             self.wrap.update_hash_cache(self.repo_dir)
         return success
 
+    def syncwrap(self) -> bool:
+        options = T.cast('SyncWrapArguments', self.options)
+        if self.wrap.type != 'git' and not self.wrap.redirected and not os.path.isdir(self.repo_dir):
+            return True
+
+        self.log(f'Sync {self.wrap.name}...')
+        if self.wrap.redirected:
+            self.wrap.filename = str(Path(self.wrap.original_filename).resolve())
+            self.wrap.parse_wrap()
+
+        if self.wrap.type != 'git':
+            return True
+
+        revision = self.wrap.get('revision').strip()
+        if revision.lower() == 'head':
+            return True
```

Comment:
Hm, not sure we should special case "head", it's not really different to wraps tracking master or any other branches. I think the point of syncwrap is to pin them on current commit/tag.